cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -34,14 +34,15 @@ import (
 )
 
 type DockerProjectOptions struct {
-	Path      string                  `yaml:"path,omitempty"      json:"path,omitempty"`
-	Context   string                  `yaml:"context,omitempty"   json:"context,omitempty"`
-	Platform  string                  `yaml:"platform,omitempty"  json:"platform,omitempty"`
-	Target    string                  `yaml:"target,omitempty"    json:"target,omitempty"`
-	Registry  osutil.ExpandableString `yaml:"registry,omitempty"  json:"registry,omitempty"`
-	Image     osutil.ExpandableString `yaml:"image,omitempty"     json:"image,omitempty"`
-	Tag       osutil.ExpandableString `yaml:"tag,omitempty"       json:"tag,omitempty"`
-	BuildArgs []string                `yaml:"buildArgs,omitempty" json:"buildArgs,omitempty"`
+	Path        string                  `yaml:"path,omitempty"      json:"path,omitempty"`
+	Context     string                  `yaml:"context,omitempty"   json:"context,omitempty"`
+	Platform    string                  `yaml:"platform,omitempty"  json:"platform,omitempty"`
+	Target      string                  `yaml:"target,omitempty"    json:"target,omitempty"`
+	Registry    osutil.ExpandableString `yaml:"registry,omitempty"  json:"registry,omitempty"`
+	Image       osutil.ExpandableString `yaml:"image,omitempty"     json:"image,omitempty"`
+	Tag         osutil.ExpandableString `yaml:"tag,omitempty"       json:"tag,omitempty"`
+	RemoteBuild bool                    `yaml:"remoteBuild,omitempty" json:"remoteBuild,omitempty"`
```

Comment:
Have you considered not exporting this as something you can set from azure.yaml? (by doing `yaml:"-"`)
Then, adding a new azd flag for `up` and `deploy` like `--build-with-acr`  ??

The idea behind that would be to let users to decide when to use remote build, instead of using the template to force the use of acr to build.

Reply:
I hadn't considered that - but I suspect that's because the thing that motivated me to make the change at first was looking at some of the existing AI samples that half use `azd` and trying to get them to use `azd` more idiomatically.  In that these cases, the sample author does not want to force a dependency on having `docker` installed, and so them opting into this via `azure.yaml` made sense to me.

But I agree that sometimes the user may want to force this to happen without having to modify the `azure.yaml` project file, and so maybe we need a gesture to allow this.

We also discussed during the show and tell perhaps trying to leverage config in some way, perhaps a `azd config set docker.remoteBuild true` or something (this would be system wide, since we don't have project wide config).

If we didn't want to use something like `--build-with-acr` we could also consider is there a way to pass config like values to an `azd` command as a one off thing?  Perhaps we could support something like `--config docker.remoteBuild true` (which would apply to everything) or something like `--config services.api.docker.remoteBuild true` (which would just apply to the API service)? 

Interested in if @kristenwomack has any additional thoughts on how we might want to service this - I still don't have a clear picture about who the best person to make this decision is - part of me feels like it is the author of the project, and so having it in the `azure.yaml` makes sense, but I do understand how there can also be a "I don't have docker and I don't want to have to edit every project file to express that intention"/

Reply:
If I picture myself as a template author, my intention is usually to provide options to the user for consuming my template.  For example, I would want to add this to my template's README:
- Running/Deploying option:
  - Install docker and run azd up
  - If you can't install docker:
    - Create a codespace and run azd up
    - If using Windows, install WSL and create a devcontainer locally, then azd up
    - Use remote build from acr (adds a few cents to your bill) by running azd up --remote-build

Then the users can decide what they want based on their needs.

That been said, as a template author, I would just describe in the Readme how to enable remote build for the template (instead of making it the deault option), like:
- You can use ARC to build containers by:
  - Update the default order for azd up adding workflow: up:  provision, deploy to azure.yaml
  - Add remoteBuild: true to aca services

Reply:
> Perhaps we could support something like --config docker.remoteBuild true (which would apply to everything) or something like --config services.api.docker.remoteBuild true (which would just apply to the API service)?

At a first glance, I really like this UX if we get requests like this. It works well if we think config is the primary happy path, and specifying a custom config is secondary path.

Conversely, I'm not sure I can think of a consistent ruling for a command specific flag like `azd up --remote-build` would ever work (how does `--remote-build` only apply to the `deploy` command?)

FWIW, I wouldn't consider these changes for now. As a template author, you could still choose not to include the default setting, and have the readme spell out the remote build enablement in azure.yaml.

Reply:
+1, if we can delay the decision of a command specific flag until we have more customer data and feedback that could allow us to understand the need.